chain-signatures/contract/src/lib.rs

```diff
@@ -373,6 +374,19 @@ impl VersionedMpcContract {
         }
     }
 
+    /// This is the derived public key of the caller given path
+    pub fn derived_public_key(&self, path: String) -> PublicKey {
+        let predecessor = env::predecessor_account_id();
```

Comment:
I think we should add in an option where people can specify the predecessor account id since the one calling this function might or might not be the right one. Like in this case:
```
user.near calls nft.near
nft.near calls sign
predecessor_id = nft.near

user.near calls derived_public_key
predecessor_id = user.near
```
```suggestion
    pub fn derived_public_key(&self, path: String, predecessor: Option<AccountId>) -> PublicKey {
        let predecessor = predecessor.unwrap_or_else(env::predecessor_account_id);
```

Reply:
sweet. Will do.